front/src/modules/pathfinding/components/IncompatibleConstraints/index.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { type FC, useEffect, useMemo, useState, useCallback } from 'react';
+
+import bbox from '@turf/bbox';
+import { feature, featureCollection } from '@turf/helpers';
+import length from '@turf/length';
+import lineSliceAlong from '@turf/line-slice-along';
+import type { LineString } from 'geojson';
+import { isArray, isNil, toPairs } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { useMap, type MapLayerMouseEvent } from 'react-map-gl/maplibre';
+
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import Collapsable from 'common/Collapsable';
+import { getMapMouseEventNearestFeature } from 'utils/mapHelper';
+
+import IncompatibleConstraintsFilters from './IncompatibleConstrainstFilters';
+import IncompatibleConstraintsInfo from './IncompatibleConstrainstInfo';
+import IncompatibleConstraintsLayer from './IncompatibleConstraintsLayer';
+import IncompatibleConstraintsList from './IncompatibleConstraintsList';
+import IncompatibleConstraintsMapFocus from './IncompatibleConstraintsMapFocus';
+import { type IncompatibleConstraintItemEnhanced, type IncompatibleConstraintTypes } from './type';
+import { getSegmentsConstraints } from './utils';
+
+interface IncompatibleConstraintsProps {
+  pathProperties?: ManageTrainSchedulePathProperties;
+}
+
+const IncompatibleConstraints: FC<IncompatibleConstraintsProps> = ({ pathProperties }) => {
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+  const map = useMap();
+
+  const [types, setTypes] = useState<Record<string, { count: number; enabled: boolean }>>({});
+  const [selected, setSelected] = useState<Set<string>>(new Set([]));
+  const [hovered, setHovered] = useState<Set<string>>(new Set([]));
```

Comment:
To improve readability ?

```suggestion
  const [selectedConstraint, setSelectedConstraint] = useState<Set<string>>(new Set([]));
  const [hoveredConstraint, setHoveredConstraint] = useState<Set<string>>(new Set([]));
```